integrations/github/highlights.scm

```diff
@@ -0,0 +1,214 @@
+  (pcal_with ("="))
+] @keyword
+
+; Literals
+(binary_number (format) @keyword)
+(binary_number (value) @number)
+(boolean) @number
+(boolean_set) @type
+(hex_number (format) @keyword)
+(hex_number (value) @number)
+(int_number_set) @type
+(nat_number) @number
+(nat_number_set) @type
+(octal_number (format) @keyword)
+(octal_number (value) @number)
+(real_number) @number
+(real_number_set) @type
+(string) @string
+(escape_char) @string.special
+(string_set) @type
+
+; Namespaces and includes
+(extends (identifier_ref) @module)
+(module name: (_) @module)
+(pcal_algorithm name: (identifier) @module)
+
+; Operators and functions
+(bound_infix_op symbol: (_) @function.builtin)
+(bound_nonfix_op (infix_op_symbol) @operator)
+(bound_nonfix_op (postfix_op_symbol) @operator)
+(bound_nonfix_op (prefix_op_symbol) @operator)
+(bound_postfix_op symbol: (_) @function.builtin)
+(bound_prefix_op symbol: (_) @function.builtin)
+(function_definition name: (identifier) @function)
+(module_definition name: (identifier) @function)
+(operator_definition name: (_) @operator)
+(pcal_macro_decl name: (identifier) @function)
+(pcal_macro_call name: (identifier) @function)
+(pcal_proc_decl name: (identifier) @function)
+(pcal_process name: (identifier) @function)
+(recursive_declaration (identifier) @operator)
+(recursive_declaration (operator_declaration name: (_) @operator))
+
+; Constants and variables
+(constant_declaration (identifier) @constant)
+(constant_declaration (operator_declaration name: (_) @constant))
+(variable_declaration (identifier) @variable.builtin)
+(pcal_var_decl (identifier) @variable.builtin)
+(pcal_with (identifier) @variable.parameter)
+((".") . (identifier) @attribute)
+(record_literal (identifier) @attribute)
+(set_of_records (identifier) @attribute)
+
+; Parameters
+(quantifier_bound (identifier) @variable.parameter)
+(quantifier_bound (tuple_of_identifiers (identifier) @variable.parameter))
+(lambda (identifier) @variable.parameter)
+(module_definition (operator_declaration name: (_) @variable.parameter))
+(module_definition parameter: (identifier) @variable.parameter)
+(operator_definition (operator_declaration name: (_) @variable.parameter))
+(operator_definition parameter: (identifier) @variable.parameter)
+(pcal_macro_decl parameter: (identifier) @variable.parameter)
+(pcal_proc_var_decl (identifier) @variable.parameter)
+
+; Delimiters
+[
+  (langle_bracket)
+  (rangle_bracket)
+  (rangle_bracket_sub)
+  "{"
+  "}"
+  "["
+  "]"
+  "]_"
+  "("
+  ")"
+] @punctuation.bracket
+[
+  ","
+  ":"
+  "."
+  "!"
+  ";"
+  (bullet_conj)
+  (bullet_disj)
+  (prev_func_val)
+  (placeholder)
+] @punctuation.delimiter
+
+; Proofs
+(proof_step_id "<" @punctuation.bracket)
+(proof_step_id (level) @number)
+(proof_step_id (name) @constant)
+(proof_step_id ">" @punctuation.bracket)
+(proof_step_ref "<" @punctuation.bracket)
+(proof_step_ref (level) @number)
+(proof_step_ref (name) @constant)
+(proof_step_ref ">" @punctuation.bracket)
```

Comment:
To be honest, proofs are quite hard to read with this highlighting:
![image](https://user-images.githubusercontent.com/15569000/152233346-4e7f9528-b00b-413d-9449-0b363455778d.png).
Highlighting `proof_step_ref (name)` the same as `proof_step_ref (level)` might improve readability. What do you think?

Reply:
I agree, which makes me sad! I put a good few days' effort to be able to tokenize the proof step IDs so the components could be highlighted independently. I'll try to figure something out. `@(constant)` might be an option. `@(label)` probably makes the most sense but it's usually highlighted the same as `@(keyword)` which makes proofs seem really samey since they have a lot of keywords.